### PR TITLE
GroupedList: add aria-expanded and aria-controls

### LINKED ISSUE #5751: GroupedList: ExpandCollapse (chevron icon ) button should have aria-expanded to announce its state
### Bug Report 
 
- __Package version(s)__: (6.38.0)
- __Browser and OS versions__: (Edge, Win 10)
 
#### Priorities and help requested:
 
Are you willing to submit a PR to fix? (No)
 
Requested priority: (High) 
 
Products/sites affected: (if applicable)
 
#### Describe the issue:
 
1. When the group by expanded or collapsed, screen reader should be able to announce its state to know the button is expanded or collapsed. 
2. In the Narrator scan mode, ExpandCollapse (chevron icon ) button is not accessible. ( the icon has role presentation & aria-hidden attributes to it, if there way to give a aria-label to the icon, then it will be accessible in scan mode)
 
#### Actual behavior:
 
ExpandCollapse (chevron icon ) button , not announcing its state. 

https://developer.microsoft.com/en-us/fabric#/components/groupedlist

### PR DESCRIPTION
**Pull request by natalieethell**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes #5751
- [X] Include a change request file using `$ npm run change`

#### Description of changes

These changes add `aria-expanded` and `aria-controls` to the expand/collapse button in the GroupHeader of GroupedList. The `aria-expanded` details whether the region the expand/collapse button controls is expanded or collapsed. The `aria-controls` specifies which region the expand/collapse button controls.

#### Justification

Taking into account the extensive discussion we had on a previous PR for this bug, and after re-reading the W3C standards site, I feel that these changes are the right approach. On the `aria-expanded` [W3C page](https://www.w3.org/WAI/GL/wiki/Using_the_WAI-ARIA_aria-expanded_state_to_mark_expandable_and_collapsible_regions), it states the following as an example of when to use the attribute:

> Example 1: Using a button to collapse and expand a region
A button hides or exposes a paragraph of text on the page. The state of the paragraph is represented by the attribute aria-expanded on the button. 

The `aria-controls` attribute seems like a necessary addition given that the list is not a direct child of the header. The W3C standards site states:

> If the element with the aria-expanded attribute controls the expansion of another grouping container that is not 'owned by' the element, the author SHOULD reference the container by using the aria-controls attribute.

#### Focus areas to test

1. Go to the GroupedList Basic Example page. 
2. Inspect the expand/collapse button. 
3. Verify that it has an `aria-expanded` attribute and that is set correctly.
4. Verify that when the group is expanded, the `aria-controls` references the id of an element that exists.
5. Collapse the group and verify that the `aria-controls` attribute disappears along with the element that was collapsed.

These changes also updated the snapshots for all of the DetailsList tests containing a `groups` prop.

### CODE REVIEW

**Comment by KevinTCoughlin** [comment]
In my testing the change re: `aria-expanded` works as expected. Here is the text output from Narrator:

My test flow for this involved:

1. Collapse Group 0
1. Expand Group 0
1. Collapse Group 0-0
1. Expand Group 0-0

```shell
Office UI Fabric React Examples - Microsoft Edge window
group 0, expand collapse group, button expanded,
Caps Lock plus Down Arrow to change to Suggestions view
expand collapse group, button collapsed,
expand collapse group, button expanded,
Caps Lock plus Down Arrow to change to Suggestions view
group 0-0, expand collapse group, button expanded,
Caps Lock plus Down Arrow to change to Suggestions view
expand collapse group, button collapsed,
expand collapse group, button expanded,
Caps Lock plus Down Arrow to change to Suggestions view
```

One _minor_ question I have before merge is re: the removal of `aria-controls` when the section is _collapsed_.

The w3 vanilla JavaScript examples I found re: collapse / expand of a tree only ever toggled `aria-expanded`. They left `aria-controls` attribute on the Element regardless of expanded state.

https://www.w3.org/WAI/GL/wiki/Using_the_WAI-ARIA_aria-expanded_state_to_mark_expandable_and_collapsible_regions#Example_1:_Using_a_button_to_collapse_and_expand_a_region

File: packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.base.tsx [hunk 1]
```diff
@@ -108,7 +109,14 @@ export class GroupHeaderBase extends BaseComponent<IGroupHeaderProps, IGroupHead
           <div className={this._classNames.dropIcon}>
             <Icon iconName="Tag" />
           </div>
-          <button type="button" className={this._classNames.expand} onClick={this._onToggleCollapse} {...expandButtonProps}>
+          <button
+            type="button"
+            className={this._classNames.expand}
+            onClick={this._onToggleCollapse}
+            aria-expanded={group ? !group.isCollapsed : undefined}
+            aria-controls={group && !group.isCollapsed ? groupedListId : undefined}
```
Comment (KevinTCoughlin) [inline packages/office-ui-fabric-react/src/components/GroupedList/GroupHeader.base.tsx]: I'm not sure that we should remove this when `group && group.isCollapsed` looking at the w3 examples:

https://www.w3.org/WAI/GL/wiki/Using_the_WAI-ARIA_aria-expanded_state_to_mark_expandable_and_collapsible_regions#Example_1:_Using_a_button_to_collapse_and_expand_a_region

Perhaps it should be:

```tsx
aria-controls={group ? groupedListId : undefined}
```

**Comment by natalieethell** [comment]
The collapsed section is completely removed from the DOM when in a collapsed state, which is why I'm removing `aria-controls` so that it doesn't reference something that no longer exists.

**Comment by KevinTCoughlin** [comment]
> 
> 
> The collapsed section is completely removed from the DOM when in a collapsed state, which is why I'm removing `aria-controls` so that it doesn't reference something that no longer exists.

Ah, that makes sense! Sorry, I should have expected the DOM.

**Pull request merged**